tools/curve_fitting/curve_fitting.xml

```diff
@@ -0,0 +1,48 @@
+<tool id="ip_curve_fitting" name="Curve Fitting" version="0.0.1" profile="20.05"> 
+    <description>to data points using (1st- or 2nd-degree) polynomial function</description>
+    <requirements>
+        <requirement type="package" version="1.20.2">numpy</requirement>
+        <requirement type="package" version="1.2.4">pandas</requirement>
+        <requirement type="package" version="1.6.2">scipy</requirement>
+    </requirements>
+    <command>
+    <![CDATA[
+         python '$__tool_directory__/curve_fitting.py'
+         '$fn_in'
+         '$fn_out'
+         '$degree'
+         '$penalty'
+         '$alpha'
+    ]]>
+    </command>
+    <inputs>
+        <param name="fn_in" type="data" format="xlsx" label="File name of input data points (xlsx)" />
+        <param name="degree" type="integer" label="Degree of the polynomial function">
+            <option value="2" selected="True">2nd degree</option>
+            <option value="1">1st degree</option>
+        </param>
+        <param name="penalty" type="select" label="Optimization objective for fitting">
+            <option value="abs" selected="True">Least absolute deviations (LAD)</option>
+            <option value="quadratic">Least squares (LS)</option>
+            <option value="student-t">Robust non-convex penalty</option>
+        </param>
+        <param name="alpha" type="float" value="0.01" label="Significance level for generating assistive curves" />
```

Comment:
Does it make sense to add max and min?

Reply:
Yes. I plan to do this in next version, because the current options/functions might be changed when making the planed workflows.